### PR TITLE
Handling password reset request

### PR DESCRIPTION
**Pull request by ykdojo**
This PR is related to #383.

It fixes part of the second issue in #383, but not the whole thing.

When I send a password reset request and click the link, this app is now able to handle it.

However, when I send a magic login request and click that link, this app does the same thing (shows you the password reset form).

It's because there doesn't seem to be any way to distinguish these two redirect links - both of them have type=recovery in them. Anyway, I'd be happy to look into the backend if it helps.

### CODE REVIEW

**Comment by ykdojo** [comment]
@awalias @kiwicopple

**Comment by ykdojo** [comment]
Note: it seems like the issue of the magic link containing type=recovery has been fixed! (by Ant)

Based on that, I'm going to test this PR again. I'm going to comment here again when it's ready to be reviewed and merged.

**Comment by ykdojo** [comment]
I went through this PR and tested it again. It's ready to be reviewed and merged now.

Next step: I think we should expose [getParameterName](https://github.com/supabase/gotrue-js/blob/dc6cf10dcac016ba4831efdb9b8683bda109dab0/src/lib/helpers.ts#L11) or something similar in Supabase's JS client so it's easier to do what this PR does.

**Comment by kiwicopple** [comment]
Ah I see what you mean. It looks like we send the user to `/?type=recovery`, and we expect the developer to parse the URL to determine the actions right?

I wonder if it's better to allow the developer to specify routes for actions in the dashboard (eg, you could specify `/password_recovery`). And if the route isn't provided we default to `/?type=recovery`

thoughts?

**Comment by ykdojo** [comment]
> Ah I see what you mean. It looks like we send the user to /?type=recovery, and we expect the developer to parse the URL to determine the actions right?
Yes.

> I wonder if it's better to allow the developer to specify routes for actions in the dashboard (eg, you could specify /password_recovery). And if the route isn't provided we default to /?type=recovery
Yes, now that I think about it, I think that would be the easiest solution for the developer.

Let me mention @awalias and @thorwebdev here to get their thoughts, too.

**Comment by ykdojo** [comment]
> I wonder if it's better to allow the developer to specify routes for actions in the dashboard (eg, you could specify /password_recovery). And if the route isn't provided we default to /?type=recovery

I've thought about this a bit more. Maybe the default should be /password_recovery#...<login_token>...?type=recovery rather than redirecting to root? I think it might be more explicit + makes it easier to understand what's going on, and it's probably closer to what most developers will want to do anyway.

**Comment by gilangmlr** [comment]
> I wonder if it's better to allow the developer to specify routes for actions in the dashboard (eg, you could specify /password_recovery). And if the route isn't provided we default to /?type=recovery

I agree. It'd be wonderful to be able to set `MAILER_URLPATHS_RECOVERY` from the dashboard, possibly the other `MAILER_URLPATHS_*` as well.

File: examples/vue3-ts-todo-list/src/App.vue [hunk 1]
```diff
@@ -27,18 +33,27 @@
 
 <script>
 import Auth from '@/components/Auth.vue'
+import PasswordReset from '@/components/PasswordReset.vue'
 import TodoList from '@/components/TodoList.vue'
 import Loading from '@/components/Loading.vue'
 import Footer from '@/components/Footer.vue'
 import { userSession, handleLogout } from '@/vuetils/useAuth'
+import { getParameterByName } from './lib/helpers'
```
Comment (gilangmlr) [inline examples/vue3-ts-todo-list/src/App.vue:41]: To make it more consistent with the other import statements.

```suggestion
import { getParameterByName } from '@/lib/helpers'
```

**Pull request merged**